### PR TITLE
Add details of each setting in the toolchain file

### PR DESCRIPTION
**Pull request by nicholasbishop**
The format of the new section is modeled after the cargo docs, for example:
https://doc.rust-lang.org/cargo/reference/profiles.html#profile-settings

This is mostly a reorganization of the existing content to make it
easier to scan for relevant details, but a few new details have also
been added:

1. For `channel`, the format is now explained in the same format as in
   concepts/toolchains.md, but with the more limited form that
   `rust-toolchain.toml` accepts. This is based on the text that was
   here previously; I haven't verified if the code is actually this
   strict.
2. The `profile` section mentions that the default profile is not
   necessarily the profile named `default`, which is subtle detail that
   can be confusing.
3. The `components` section notes that the list is additive with the
   profile's list of components.
4. The `targets` field is now documented, previously it wasn't mentioned
   other than to say it's ignored with `path`.

### CODE REVIEW

File: doc/src/overrides.md
Comment (0xPoe) [inline doc/src/overrides.md]: If you use other actions like https://github.com/actions-rs/toolchain#inputs, it will be `default` instead of minimal. So I prefer not to mention this here.

Comment (nicholasbishop) [reply doc/src/overrides.md]: OK, I have removed that sentence.

**Review by 0xPoe — APPROVED** [review]
Looks good to me. Thanks for your patch!
But I am not a native English speaker, so I'd like to wait for approval from @kinnison or @rbtcollins before we merge it.

**Comment by 0xPoe** [comment]
You can check out this [rendered page](https://github.com/nicholasbishop/rustup/blob/bishop-specify-additive/doc/src/overrides.md).

**Comment by nicholasbishop** [comment]
Friendly ping, would love to get this merged :)

**Comment by 0xPoe** [comment]
I am going to merge this PR. If we find some grammar issues we can fix them later. Thanks! Sorry for the long wait.

**Pull request merged**
